substrate/client/src/client.rs

```diff
@@ -82,6 +94,32 @@ pub enum BlockStatus {
 	Unknown,
 }
 
+/// Block data origin.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum BlockOrigin {
+	/// Genesis block built into the client.
+	Genesis,
+	/// Block is part of the initial sync with the network.
+	NetworkInitialSync,
+	/// Block was broadcasted on the network.
+	NetworkBroadcast,
```

Comment:
aren't all blocks broadcast over the network in some sense? doesn't it make more sense to provide hints about whether a block is the head of the chain or not?

Reply:
The head hint is provided as a separate parameter to `import` function. This is for something else. Mainly for logging and possibly skipping certain checks for local blocks.